api/lib/round-tracker.js

```diff
@@ -238,14 +242,30 @@ export async function maybeCreateSparkRound (pgClient, {
   const { rowCount } = await pgClient.query(`
     INSERT INTO spark_rounds
     (id, created_at, meridian_address, meridian_round, start_epoch, max_tasks_per_node)
-    VALUES ($1, now(), $2, $3, $4, $5)
+    VALUES (
+      $1,
+      now(),
+      $2,
+      $3,
+      $4,
+      COALESCE(
+        (SELECT max_tasks_per_node FROM spark_rounds WHERE meridian_round = $3 - 1),
+        $5
+      )
+        * $6
+        / COALESCE(
+            (SELECT measurement_count FROM spark_rounds WHERE meridian_round = $3 - 1),
+            $6
+          )
+    )
```

Comment:
Verified in https://onecompiler.com/postgresql/42rp72e93:

1. Create round 1, `max_tasks_per_node` will be set to `BASELINE_MAX_TASKS_PER_NODE`
2. Set `measurement_count` to 250k, half of the desired 500k
3. Create round 2, `max_tasks_per_node` will now be double of the previous round.

```
 meridian_round | max_tasks_per_node | measurement_count 
----------------+--------------------+-------------------
              1 |                 15 |            250000
              2 |                 30 |                 0
(2 rows)
```

Reply:
@bajtos I believe we can refactor this query so it doesn't run two selects on the same table. Otoh, I believe this code path isn't too hot to require this. But, if we would want to do so, @bajtos do you know how to refactor this query?

One option is to move the selects out of this query and into the `if (rowCount)` below.

Reply:
Would something like this work?

```sql
WITH prev_round AS (
  SELECT max_tasks_per_node, measurement_count
  FROM spark_rounds WHERE meridian_round = $3 - 1
)
INSERT INTO spark_rounds
  -- (..)
  COALESCE (prev_round.max_task_per_node, $5)
    * $6 
    / COALESCE (prev_round.measurement_count, $6)
```

You may need to convert a the `prev_round` rowset into a scalar, I did something similar recently but don't remember details.

Reply:
according to https://stackoverflow.com/a/50364985:

> If the source of an insert statement is a select do not use the VALUES keyword.

But, we do need it here 🤔

Reply:
It failed in onecompiler as well, saying there's a missing `FROM` clause for `prev_round`